src/Tree/index.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react'
+import propTypes from 'prop-types'
+import cx from 'classnames'
+import css, { resolve } from 'styled-jsx/css'
+
+import { ArrowDown } from '../icons/Arrow'
+import { colors } from '../theme'
+
+const Label = ({ icon, children }) => (
+    <div>
+        {children}
+
+        <style jsx>{`
+            div {
+                min-height: 24px;
+                padding: 3px 0;
+            }
+        `}</style>
+    </div>
+)
+
+const Contents = ({ children, open }) => (
+    <div className={cx({ open })}>
+        {children}
+
+        <style jsx>{`
+            div {
+                height: 0;
+                overflow: hidden;
+                flex-grow: 1;
+            }
+
+            .open {
+                height: auto;
+            }
+        `}</style>
+    </div>
+)
+
+const arrowStyle = resolve`
+    div > :global(svg){
+        fill: #e2e2e2;
+    }
+`
+
+const Arrow = ({ hasLeafes, open, onToggleOpen, arrowTopOffset }) => {
+    const arrowIcon = hasLeafes ? <ArrowDown /> : <span />
+
+    return (
+        <div
+            className={cx('arrow', arrowStyle.className, {
+                open,
+                'has-leafes': hasLeafes,
+            })}
+        >
+            <span onClick={onToggleOpen}>{arrowIcon}</span>
+
+            <style jsx>{`
+                div {
+                    width: 24px;
+                    position: relative;
+                    flex-shrink: 0;
+                }
+
+                .has-leafes.open:after {
+                    content: '';
+                    background: ${colors.grey400};
+                    height: calc(100% - 24px);
+                    left: 12px;
+                    position: absolute;
+                    top: 15px;
+                    width: 1px;
+                    z-index: 1;
+                }
+
+                span {
+                    display: block;
+                    padding-top: ${arrowTopOffset};
+                    position: relative;
+                    transform: rotate(-90deg);
+                    z-index: 2;
+                    fill: ${colors.grey700};
+                }
+
+                .open span {
+                    transform: rotate(0);
+                }
+            `}</style>
+            {arrowStyle.styles}
+        </div>
+    )
+}
+
+const Content = ({ children }) => <div>{children}</div>
+
+export const Tree = ({
+    children,
+    hasLeafes,
+    open,
+    onToggleOpen,
+    arrowTopOffset,
+}) => {
+    const className = cx('tree', {
+        open,
+        'has-leafes': hasLeafes,
+    })
+
+    return (
+        <div className={className}>
+            <Arrow
+                open={open}
+                hasLeafes={hasLeafes}
+                onToggleOpen={onToggleOpen}
```

Comment:
Does this fire on toggle or on open? I would prefer either `onToggle` or `onOpen`, respectively.

Reply:
Looking at it now, I think it should be `onClick`.

Reply:
well it's basically toggling the open state.. one could assume that `onToggle` refers to the expanded/collapsed state, so I guess it might work, but `onToggleOpen` is def. more precise

Reply:
I see what you mean about it toggling a state.

The name `onToggleOpen` implies to me that it only fires when it is opened, but when I check the implementation it's bound to the `onClick` handler so it will fire every time; not just when it goes from closed to open. That's what I find confusing.

Reply:
Either way, might change now after the discussion on Slack anyway so we'll see how it flies. :)

Reply:
I think we could split it into `onExpand` and `onCollapse`, the names are much better and allow for finer control, I'd say

Reply:
Splitting them sounds good to me. We use `onOpen` and `onClose` else-where  in `ui-core`, would those terms be appropriate here too? 

If possible I want to make the API be as consistent as possible and not use new terminology in each component.